pygly/alignment.py

```diff
@@ -690,11 +855,55 @@ def subtree_leq(self,m, tg,root=True):
             if not self.monoleq(m, tg):
                 return False
 
-        for tg_linkset in choose(tg.links(), len(m.links())):
-            for ii,jj in itermatchings(m.links(), tg_linkset,
-                                       lambda i, j: self.linkleq(i, j) and self.subtree_leq(i.child(), j.child(), root=False)):
+
+        assert len(filter(lambda l:l.repeat_bridge_link(), m.links())) in [0, 1]
+        mlinks_no_repeat_path = m.links_without_repeat()
+        mlinks_repeat_path    = m.links_with_repeat()
+
+
+
+        tglinks = tg.links()
+        tglinks_type_1 = []
+        tglinks_type_2 = []
+        tglinks_type_3 = []
+
+        for l in tglinks:
+
+            if l.basic_link():
+                tglinks_type_1.append(l)
+
+            elif l.repeat_bridge_link():
+                tglinks_type_2.append(l)
+
+            elif l.repeat_unit_out_link():
+                tglinks_type_3.append(l)
+
+        assert len(tglinks_type_2) in [0, 1]
+        tglinks_no_repeat_path = tglinks_type_1 + tglinks_type_3
+        tglinks_repeat_path = tglinks_type_1 + tglinks_type_2
+
+
+
+        for tglinks_no_repeat_path_partial in choose(tglinks_no_repeat_path, len(mlinks_no_repeat_path)):
```

Comment:
Something is wrong here, when I do compute (glycomotif) alignments it appears to go into a 50-60 deep choose recursion. The specific accessions appear to be G34111ZG (glycan) and G46392BZ (motif), although when I do this comparison by itself, I don't have the same issue. Note that G34111ZG has a substituent in link.

Reply:
I removed this if statement (code back the way it was) which permits structures with substituent in links to be cloned. Why was this check added?